pkg/kubelet/volumes.go

```diff
@@ -87,11 +91,22 @@ func (vh *volumeHost) NewWrapperUnmounter(volName string, spec volume.Spec, podU
 		// Not found but not an error
 		return nil, nil
 	}
+
 	c, err := plugin.NewUnmounter(spec.Name(), podUID)
-	if err == nil && c == nil {
-		return nil, errUnsupportedVolumeType
+	if err != nil {
+		return nil, err
+	}
+	if c != nil {
+		// return good volume unmounter
+		return c, nil
 	}
-	return c, nil
+
+	// generic volume type error
+	return nil, createVolumeTypeError(volName)
+}
+
+func createVolumeTypeError(volName string) error {
+	return fmt.Errorf("Unable to mount volume or unsupported volume type for volume: %s for plugin: Unknown - check logs or pod describe events for possible root cause", volName)
```

Comment:
We need to expose the actual error wherever possible; don't eat errors, and don't add vague text to errors.  Think about it from the user's perspective -- we want to give them the information most relevant to solving the problem.